classes/StaffManager.php

```diff
@@ -0,0 +1,575 @@
+                return false;
+            }
+            $unban_string = ucwords($unban_type) . " Ban";
+            unset($ban_data[$unban_type]);
+        }
+
+        $ban_data = empty($ban_data) ? '' : json_encode($ban_data);
+
+        $this->system->query("UPDATE `users` SET `ban_data`='{$ban_data}' WHERE `user_id`='{$user_data['user_id']}' LIMIT 1");
+        if($this->system->db_last_affected_rows) {
+            $this->addRecord($user_data['user_id'], $user_data['user_name'], self::RECORD_BAN_REMOVED,
+                "Removed " . ucwords($unban_type) . " Ban.", false);
+            $this->staffLog(StaffManager::STAFF_LOG_HEAD_MOD,
+                "($this->user_name} ({$this->user_id}) removed {$user_data['user_name']}\'s ({$user_data['user_id']}) "
+                . $unban_string . ".");
+            return true;
+        }
+        return false;
+    }
+
+    public function sendOW($content, $user_data) {
+        $this->system->query("INSERT INTO `official_warnings` 
+            (`staff_id`, `staff_name`, `user_id`, `time`, `data`, `viewed`)
+            VALUES 
+            ('{$this->user_id}', '{$this->user_name}', '{$user_data['user_id']}', '" . time() . "', '{$content}', 0)
+        ");
+        if($this->system->db_last_insert_id) {
+            $this->addRecord($user_data['user_id'], $user_data['user_name'], self::RECORD_OFFICIAL_WARNING, $content, false);
+            $this->staffLog(self::STAFF_LOG_MOD, "{$this->user_name}($this->user_id}) sent {$user_data['user_name']}({$user_data['user_id']}) an Official Warning.");
+            return true;
+        }
+        return false;
+    }
+
+    public function removeProfileData($type, $user_data) {
+        $query = "UPDATE ";
+        switch($type) {
+            case self::BAN_TYPE_JOURNAL:
+                $query .= "`journals` SET `journal`='' WHERE `user_id`='{$user_data['user_id']}' LIMIT 1";
+                break;
+            case self::BAN_TYPE_AVATAR:
+                $query .= "`users` SET `avatar_link`='./images/default_avatar.png' WHERE `user_id`='{$user_data['user_id']}' LIMIT 1";
+                break;
+            default:
+                return false;
+        }
+        $this->system->query($query);
+        if($this->system->db_last_affected_rows) {
+            $this->staffLog(self::STAFF_LOG_MOD, "Removed {$user_data['user_name']}({$user_data['user_id']})\'s "
+            . ucwords($type) . ".");
+            return true;
+        }
+        return false;
+    }
+
+    public function getStaffLevelName($staff_level = null, $name_type = 'short') {
+        if($staff_level === null) {
+            return self::$staff_level_names[$this->staff_level][$name_type];
+        }
+        return self::$staff_level_names[$staff_level][$name_type];
+    }
+
+    public function getBanLengths() {
+        $ban_lengths = [
+            1 => '1 Day',
+            7 => '1 Week',
+            30 => '1 Month',
+            90 => '3 Months'
+        ];
```

Comment:
Ideally let's move this to a const or static variable at the top of the class so it's easier to find and edit